src/DonationSummary/resources/js/summary.js

```diff
@@ -100,50 +118,51 @@ window.GiveDonationSummary = {
     },
 
     /**
-     * @since 2.17.0
-     */
-    initFees: function () {
-        GiveDonationSummary.observe('.give_fee_mode_checkbox', GiveDonationSummary.handleFees);
-    },
-
-    /**
+     * @unreleased Remove dependency on checkbox. Removed first argument.
      * @since 2.18.0
      */
-    handleFees: function (targetNode, $form) {
-        $form.find('.fee-break-down-message').hide();
-        $form.find('.js-give-donation-summary-fees').toggle(targetNode.checked);
+    handleFees: function ($form) {
+        const feeModeEnableElement = $form.find('[name="give-fee-mode-enable"]');
 
-        if (!targetNode.checked) {
+        if (!feeModeEnableElement || 'true' !== $form.find('[name="give-fee-mode-enable"]').val()) {
+            $form.find('.js-give-donation-summary-fees').toggle(false);
             return;
         }
 
-        // Hack: (Currency Switcher) The total is always stored using a the decimal separator as set by the primary currency.
-        const formData = new FormData($form[0]);
-        const fee = formData.get('give-fee-amount');
-        $form.find('[data-tag="fees"]').html(GiveDonationSummary.format_amount(fee, $form));
+        $form.find('.js-give-donation-summary-fees').toggle(true);
+
+        const feeMessageTemplateParts = $form.find('.give-fee-message-label').attr('data-feemessage').split(' ');
+        const feeMessageParts = $form.find('.give-fee-message-label-text').text().split(' ');
+        const formattedFeeAmount = feeMessageParts
+            .filter((messagePart) => !feeMessageTemplateParts.includes(messagePart))
+            .pop();
+        $form.find('[data-tag="fees"]').html(formattedFeeAmount);
     },
 
     /**
      * @since 2.17.0
      */
     initTotal: function () {
+        const handleFees = this.handleFees;
+
         GiveDonationSummary.observe('.give-final-total-amount', function (targetNode, $form) {
-            // Hack: (Currency Switcher) The total is always stored using a the decimal seperator as set by the primary currency.
-            const total = targetNode.dataset.total.replace('.', Give.form.fn.getInfo('decimal_separator', $form));
-            $form.find('[data-tag="total"]').html(GiveDonationSummary.format_amount(total, $form));
+            $form.find('[data-tag="total"]').html(targetNode.textContent);
+
+            handleFees($form);
```

Comment:
Same here, we don't need to have a local instance of `const handleFees = this.handleFees;`.

```suggestion
            GiveDonationSummary.handleFees($form);
```